File: OJTrain/Python/leetcode3.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        max_win_size = 0
        for i in range(0, len(s)):
            j = i
            exist_chars = set()
            while j < len(s):
                if s[j] in exist_chars:
                    break
                exist_chars.add(s[j])
                j += 1
            max_win_size = max(j - i, max_win_size)
        return max_win_size


solution = Solution()
print(solution.lengthOfLongestSubstring(" "))

File: OJTrain/Go/src/leetcode3.go
package main

func max(x, y int) int {
	if x > y {
		return x
	}
	return y
}

func lengthOfLongestSubstring(s string) int {
	maxWinSize := 0
	for i := 0; i < len(s); i++ {
		existChars := make(map[byte]bool)
		for j := i; j < len(s); {
			if existChars[s[j]] {
				break
			}
			existChars[s[j]] = true
			j += 1
			maxWinSize = max(j-i, maxWinSize)
		}
	}
	return maxWinSize
}
